arm-templates/base/deploy-aks.json

```diff
@@ -103,16 +104,19 @@
             "properties": {
                 "kubernetesVersion": "[parameters('kubernetesVersion')]",
                 "enableRBAC": true,
+                "nodeResourceGroup": "[if(contains(resourceGroup().name, 'dev'), concat(variables('clusterName'), '-dataplane'), '')]", //custom nodepool rg for dev only
                 "dnsPrefix": "[concat(variables('clusterName'),'-dns')]",
                 "agentPoolProfiles": [
                     {
                         "name": "agentpool",
+                        "mode": "System",
                         "count": 2,
+                        "osDiskType": "[if(contains(resourceGroup().name, 'dev'), 'Ephemeral', 'Managed')]", //Epheremal only for dev
                         "vmSize": "Standard_DS12_v2",
                         "osType": "Linux",
                         "enableAutoScaling": "[parameters('enableAutoscaler')]",
-                        "minCount": "[parameters('minNodes')]",
-                        "maxCount": "[parameters('maxNodes')]",
+                        //"minCount": "[parameters('minNodes')]",
```

Comment:
Currently autoscaler disabled through CLI for dev and enabled for prod. If "enableAutoScaling" = false, then minCount and maxCount cannot be set.

Reply:
autoscaler disabled is required for "az aks stop" command for dev cluster while the command is in preview.